freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java

```diff
@@ -151,4 +155,20 @@ private String readCatalogFromFile(String catalogUrl) throws IOException {
         File customCatalogFile = new File(etcConfigDir, catalogUrl);
         return FileReaderUtils.readFileFromPath(customCatalogFile.toPath());
     }
+
+    private Versions getVersions(ImageCatalog catalog) {
+        if (catalog.getVersions() == null || catalog.getVersions().getFreeIpaVersions() == null) {
+            LOGGER.debug("FreeIPA versions are missing from the image catalog.");
+            List<String> advertisedImageUuids = catalog.getImages().getFreeipaImages().stream()
+                    .filter(Image::isAdvertised)
+                    .map(Image::getUuid)
+                    .collect(Collectors.toList());
+            List<FreeIpaVersions> versionList = List.of(new FreeIpaVersions(List.of(freeIpaVersion), List.of(), advertisedImageUuids));
+            Versions versions =new  Versions(versionList);
```

Comment:
Just a nitpick, but "=new" looks weird, for a second it seriously made me wonder if it's a standalone unary operator I've never heard of :) Probably freaks out Checkstyle as well, doesn't it?

Reply:
Hmm, its wired, I can remember I have fixed this by the checkstyle failed locally...